cartographer/mapping_3d/optimizing_local_trajectory_builder.cc

```diff
@@ -223,6 +237,9 @@ OptimizingLocalTrajectoryBuilder::MaybeOptimize(const common::Time time) {
   }
 
   ceres::Problem problem;
+  const Submap* const matching_submap =
+      submaps_->Get(submaps_->matching_index());
+  TransformStates(matching_submap->local_pose().inverse());
```

Comment:
This is quite hard to understand. Add a comment that we in-place transform the state into submap frame from odom frame since the optimization problem expects it like that and that we revert this after we are done optimizing.